### DISCUSSION TITLE
Bit-Perfect / No volume is ON but am able to change volume! Not Working?

### DISCUSSION THREAD

**Discussion by Roland-B** [rern/rAudio#442] [Fixed Issues] [answered]
@rern I am using 64-Bit version (not tested this issue on other versions yet) and have bit perfect on with no rotary volume control showing but discovered with an accidental click below the cover that I was able to change volume. Surely this shouldn't be able to be done if the mixer is off and is "Bit-Perfect"?

![rune1a](https://user-images.githubusercontent.com/43496391/126355385-1cb3feca-6afe-4e61-a5ea-0a5518c3c658.jpg)
 
![rune1b](https://user-images.githubusercontent.com/43496391/126355442-4b5a2e06-ea35-4337-83af-b886785b065c.jpg)

rAudio 1 • 20210716
5.10.50-1-ARCH aarch64
Raspberry Pi 4 Model B Rev 1.2
BCM2711 • 3.7GiB
4 @ 1.5GHz

**Comment by R101G** [comment]
I am using the same OS version with the same settings, but with a different (XMOS) USB interface, and I can't reproduce this problem.
Clicking on the mixer device settings shows a volume control which is locked at 100%.

**Reply by Roland-B** [reply]
@R101-Github  Thanks for trying. It is not an exactly an easy thing to do if not sure where to click, you have to click on just the right spot below the album cover.

My mixer device settings also show a volume control locked at 100% but in effect is is not.. for me anyhow.. if it's possible to change the volume via the main playing page then its not bypassing mixer etc. but just giving the illusion of doing it.

Edit: Just tried a old non 64-bit image and although changed volume under cover again (using hidden controls) it had no audible effect on volume. 

@R101-Github - Did you get the same effect using your 64-Bit build.. volume bar moved and no change or did you not get as far as using the hidden control at all? Thanks.

Roland.

**Reply by R101G** [reply]
@Roland-B 
You are right. Clicking at the top of the cover art actually showed the volume control. You have to click exactly on the bottom edge of the art to see it otherwise.

Playing with the level control muted the output, and it's taken a while to get the sound back. A mute button appeared below the album name, and a bit of fiddling eventually restored the sound.

**Reply by Roland-B** [reply]
@R101-Github Thanks for trying again, much appreciated. I have re-imaged the 64-Bit Build and the bug is also on latest version I have tested. I shall try a fresh latest build of non 64-bit build and report back, although im pretty sure is only on the 64-bit version.

Roland.

**Reply by R101G** [reply]
@Roland-B  I have two USB interfaces, both XMOS based. The JLS interface shows "not available" in the mixer field under Player settings. The phantom volume control will show a change in level but this has no effect on the actual  output, which is still bit-perfect. I know this because it is attached to an HDCD filter chip and the indicator lights up when it should.

The DIYINHK interface shows a mixer entry in Player settings, and the phantom volume certainly affects the output. Whether it is bit-perfect when 100% I do not know, as this DAC does not have a PM filter chip. I can knock something up if we need to be sure, though.

I hope this helps rern eliminate the problem. I have accidentally invoked the phantom control several times now!

**Reply by Roland-B** [reply]
@R101-Github Strange is OK for one and not the other? My understanding is if the player (not DAC) is changing the volume then it's also changing the "bits". Only the DAC itself can change the volume and leave bit pefect if it has volume functionality, but have no proof other than what ive read previously over the years. Anything you can put together to prove the issue would be great (I know I probably couldn't do also) but hopefully not required if is an easy thing for @rern to find and fix.

Yup, same here.. I have accidentally hit a few times when using tablet and not paying attention to what im doing.. slightly annoying to say the least :) Not sure why, but do love the term you came up for it as "Phantom Control".

Roland.

**Reply by R101G** [reply]
@Roland-B I assume the drivers are different, and this allows some rogue UI code to become a nuisance. If you run the alsamixer command, you can see the unwanted volume control is altering the level there. If the control did not appear, the ALSA level would stay at 100% and there would be no problem.

**Comment by Roland-B** [comment, marked as answer]
Looks to be fixed with todays update (**20210704**), although did require me to change audio to on-board headphones and then back to "E30" DAC for the change to take place. Thanks @rern for remediating the bug and @R101-Github for helping confirm the issue. I will wait for confirmation from R101-Github before closing.

Roland B

**Comment by R101G** [comment]
@Roland-B
 Looks good to me too, and I can confirm the output is bit-perfect with both my interfaces. 

@rern 
Presumably, whether a mixer entry is shown or not depends on the XMOS firmware?
